android/app/src/main/java/com/expensify/chat/customairshipextender/NotificationCache.java

```diff
@@ -0,0 +1,168 @@
+package com.expensify.chat.customairshipextender;
+
+import android.content.Context;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.util.Base64;
+
+import androidx.core.app.Person;
+import androidx.core.graphics.drawable.IconCompat;
+
+import com.expensify.chat.MainApplication;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.HashMap;
+
+public class NotificationCache {
+
+    private static final String CACHE_FILE_NAME = "notification-cache";
+    private static HashMap<String, NotificationData> cache = null;
+
+    /*
+     * Get NotificationData for an existing notification or create a new instance
+     * if it doesn't exist
+     */
+    public static NotificationData getNotificationData(long reportID) {
+        if (cache == null) {
+            cache = readFromInternalStorage();
+        }
+
+        NotificationData notificationData = cache.get(Long.toString(reportID));
+
+        if (notificationData == null) {
+            notificationData = new NotificationData();
+            setNotificationData(reportID, notificationData);
+        }
+
+        return notificationData;
+    }
+
+    /*
+     * Set and persist NotificationData in the cache
+     */
+    public static void setNotificationData(long reportID, NotificationData data) {
+        if (cache == null) {
+            cache = readFromInternalStorage();
+        }
+
+        cache.put(Long.toString(reportID), data);
+        writeToInternalStorage();
+    }
+
+    private static void writeToInternalStorage() {
```

Comment:
Actually, it's much better if we can keep the Context ref alive only for the duration of the function: 

```
private static void writeToInternalStorage(Context context) {
```